### PR TITLE
Add typings to OCP\Calendar\ICalendar and implementation

### PR DESCRIPTION
**Pull request by tcitworld**

### CODE REVIEW

File: lib/public/Calendar/ICalendar.php [hunk 1]
```diff
@@ -68,11 +68,11 @@ public function getDisplayColor();
 	 * @return array an array of events/journals/todos which are arrays of key-value-pairs
 	 * @since 13.0.0
 	 */
-	public function search($pattern, array $searchProperties = [], array $options = [], $limit = null, $offset = null);
+	public function search(string $pattern, array $searchProperties = [], array $options = [], ?int $limit = null, ?int $offset = null): array;
 
 	/**
 	 * @return integer build up using \OCP\Constants
```
Comment (ChristophWurst) [inline lib/public/Calendar/ICalendar.php:74]:
```suggestion
	 * @return int build up using \OCP\Constants
```

File: apps/dav/lib/CalDAV/CalendarImpl.php [hunk 2]
```diff
@@ -115,7 +108,7 @@ public function search($pattern, array $searchProperties = [], array $options =
 	 * @return integer build up using \OCP\Constants
```
Comment (ChristophWurst) [inline apps/dav/lib/CalDAV/CalendarImpl.php:108]:
```suggestion
	 * @return int build up using \OCP\Constants
```

**Review by ChristophWurst — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
:shipit:

File: apps/dav/lib/CalDAV/CalendarImpl.php [hunk 3]
```diff
@@ -46,14 +46,9 @@
...
-	private $backend;
-
-	/** @var Calendar */
-	private $calendar;
-
-	/** @var array */
-	private $calendarInfo;
+	private CalDavBackend $backend;
+	private Calendar $calendar;
+	private array $calendarInfo;
```
Comment (come-nc) [inline apps/dav/lib/CalDAV/CalendarImpl.php:51]: If possible add a phpdoc with what type is in the array.
Something like `@var array<string, ?string>`

File: lib/public/Calendar/ICalendar.php [hunk 4]
```diff
@@ -49,30 +49,30 @@ public function getUri(): string;
...
 	 * 	['timerange' => ['start' => new DateTime(...), 'end' => new DateTime(...)]]
-	 * @param integer|null $limit - limit number of search results
-	 * @param integer|null $offset - offset for paging of search results
+	 * @param int|null $limit - limit number of search results
+	 * @param int|null $offset - offset for paging of search results
 	 * @return array an array of events/journals/todos which are arrays of key-value-pairs
 	 * @since 13.0.0
 	 */
-	public function search($pattern, array $searchProperties = [], array $options = [], $limit = null, $offset = null);
+	public function search(string $pattern, array $searchProperties = [], array $options = [], ?int $limit = null, ?int $offset = null): array;
```
Comment (come-nc) [inline lib/public/Calendar/ICalendar.php:71]: Add this typing in the implementation as well

**Review by come-nc — APPROVED** [review]

**Review by miaulalala — APPROVED** [review]
LGTM

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
```
There was 1 error:
221	
222	1) OCA\DAV\Tests\unit\CalDAV\CalendarManagerTest::testSetupCalendarProvider
223	TypeError: Return value of OCA\DAV\CalDAV\CalendarImpl::getKey() must be of the type string, int returned
224	
225	/drone/src/apps/dav/lib/CalDAV/CalendarImpl.php:67
226	/drone/src/apps/dav/tests/unit/CalDAV/CalendarManagerTest.php:85
227	/drone/src/apps/dav/lib/CalDAV/CalendarManager.php:79
228	/drone/src/apps/dav/lib/CalDAV/CalendarManager.php:66
229	/drone/src/apps/dav/tests/unit/CalDAV/CalendarManagerTest.php:96
```

**Comment by tcitworld** [comment]
Fixed.

**Comment by come-nc** [comment]
CI failure unrelated

**Pull request merged**